internal/graph/cached_resolver.go

```diff
@@ -175,24 +175,25 @@ func (c *CachedCheckResolver) ResolveCheck(
 // cache key. If the contextual tuples are different order, it is possible that a different
 // cache key will be produced. This will result in duplicate entries.
 func checkRequestCacheKey(req *ResolveCheckRequest) (string, error) {
-	var b bytes.Buffer
-	if err := gob.NewEncoder(&b).Encode(req.GetTupleKey()); err != nil {
-		return "", err
-	}
+	var contextualTuplesCacheKey string
 
-	tupleCacheKey := b.String()
+	contextualTuples := req.GetContextualTuples()
 
-	var c bytes.Buffer
-	if err := gob.NewEncoder(&c).Encode(req.GetContextualTuples()); err != nil {
-		return "", err
-	}
+	if len(contextualTuples) > 0 {
+		var c bytes.Buffer
 
-	contextualTuplesCacheKey := c.String()
+		// only use gob if there are contextual tuples as it is CPU intensive
+		if err := gob.NewEncoder(&c).Encode(req.GetContextualTuples()); err != nil {
+			return "", err
+		}
+
+		contextualTuplesCacheKey = c.String()
+	}
 
 	key := fmt.Sprintf("%s/%s/%s/%s",
```

Comment:
Seems we should avoid including the trailing `/` if the contextual tuples slice is empty.

`store/model/tuple` vs `store/model/tuple/`

Reply:
Keeping slash allows us to add more fields (maybe ABAC's condition) if needed.

Reply:
We can add the slash at that time then. For now it's not needed.